qa/rpc-tests/graphene_optimized.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Unlimited developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+
+from grapheneblocks import GrapheneBlockTest
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+
+class GrapheneOptimizedTest(GrapheneBlockTest):
+    
+    def setup_network(self, split=False):
+        node_opts = [
+            "-rpcservertimeout=0",
+            "-debug=graphene",
+            "-use-grapheneblocks=1",
+            "-use-thinblocks=0",
+            "-compute-optimize-graphene=1",
+            "-excessiveblocksize=6000000",
+            "-blockprioritysize=6000000",
+            "-blockmaxsize=6000000"]
+
+        self.nodes = [
+            start_node(0, self.options.tmpdir, node_opts),
+            start_node(1, self.options.tmpdir, node_opts),
+            start_node(2, self.options.tmpdir, node_opts)
+        ]
+
+        interconnect_nodes(self.nodes)
+        self.is_network_split = False
+        self.sync_all()
+
```

Comment:
seems this test only connect the nodes but doesn't actually create or propagate any blocks.  I would suggest two things, mine blocks, create transactions and then mine another block so that we can be sure we've propagated a graphene block...futhermore, make sure to turn off compactblock propagation just to be sure.

Also, you should have one node which has grapene on but computeoptimize off, so that we test that condition as well.

Reply:
It's not very obvious, but class `GrapheneOptimizedTest` inherits from `GrapheneBlockTest` (the old test), so actually `graphene_optimized.py` runs the same test as `grapheneblocks.py` only with a different configuration for the nodes (specifically they all set compute-optimize-graphene=1).

With that said, I like your idea to test interoperability between nodes with and without compute optimization. So I've modified the test to have one node set the flag to 0. I also added a log statement in graphene_set.cpp to ensure that both types of Bloom filters are exercised by this test.